### PR TITLE
NPM Only Components

### PR DESCRIPTION
**Pull request by notlee**
- Deprecates the Origami specification
- Adds guides/comms for the transition to npm-only releases

Build on the the following PRs:
- https://github.com/Financial-Times/origami-website/pull/357
- https://github.com/Financial-Times/origami-website/pull/359
- https://github.com/Financial-Times/origami-website/pull/357

Todo (add to this list):

- [x] A blog post to describe why the spec is being deprecated, and what will happen next. Update TODO links to this.
~- [ ] [Write npm 6 migration guide](https://github.com/Financial-Times/origami-website/pull/360#discussion_r639894083). (Check for npm 7 references which are outdated).~ - We don't support npm 6 and are not provided written guidance on how to get origami to work with npm 6

### CODE REVIEW

**Comment by notlee** [comment]
Converted to a draft as we still have docs/comms to write

File: _posts/2021-07-01-origami-2.0-and-how-to-migrate.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+# Plans for the future
+
+Origami used to be versioned only with a major version (Origami 1), this made it hard to track any of the changes being made to Origami which did not warrant incrementing the major version. Now we have incremented the major version, we thought it was a good time to include minors in the versioning scheme. This will make it simpler to make more frequent changes to Origami as a whole and have a record of what thoses changes were and when they were made.
+
+We are looking for people's thoughts on what Origami is missing and/or how Origami could be reimagined to help better achieve it's aim to unify and document the style and experience across the digital products of every FT brand. Some of the parts of Origami we are wanting to improve are around how the documentation and demonstrations for the components are built. Currently they are built in isolation of one another, we would like to see if building them together would lead to a better documented and test set of components.
+
+# First Origami 2.0 Component
```
Comment (JakeChampion) [inline _posts/2021-07-01-origami-2.0-and-how-to-migrate.md:29]:
```suggestion
# First npm-only Origami Component
```

Comment (notlee) [reply _posts/2021-07-01-origami-2.0-and-how-to-migrate.md:29]: Thanks, I'll do this as a commit to update the file name and a couple of other references

File: _posts/2021-07-01-origami-on-npm-and-how-to-migrate.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+# Plans for the future
+
+Origami used to be versioned only with a major version (Origami 1), this made it hard to track any of the changes being made to Origami which did not warrant incrementing the major version. Now we have incremented the major version, we thought it was a good time to include minors in the versioning scheme. This will make it simpler to make more frequent changes to Origami as a whole and have a record of what thoses changes were and when they were made.
+
+We are looking for people's thoughts on what Origami is missing and/or how Origami could be reimagined to help better achieve it's aim to unify and document the style and experience across the digital products of every FT brand. Some of the parts of Origami we are wanting to improve are around how the documentation and demonstrations for the components are built. Currently they are built in isolation of one another, we would like to see if building them together would lead to a better documented and test set of components.
+
+# First Origami npm-only Origami Component
```
Comment (JakeChampion) [inline _posts/2021-07-01-origami-on-npm-and-how-to-migrate.md:29]:
```suggestion
# First npm-only Origami Component
```

File: _tutorials/bower-to-npm.md [hunk 3]
```diff
@@ -28,15 +28,17 @@ _Note: These [drawbacks will be addressed with component updates by Q3 2021](/bl
 ## Migrating a project which uses Origami components from bower to npmjs
 
 ### TLDR
```
Comment (JakeChampion) [inline _tutorials/bower-to-npm.md:30]: Instead of having this page, what do we think about redirecting to the new migration guide? https://github.com/Financial-Times/origami-website/blob/cb595392b2a43dee22755589b277ed9a455afe69/_posts/2021-07-01-origami-on-npm-and-how-to-migrate.md#using-origami-via-bower

File: _tutorials/manual-build.md [hunk 4]
```diff
@@ -90,38 +90,23 @@ Let's head over to <a href="https://registry.origami.ft.com/components/o-table#d
...
-<pre><code class="o-syntax-highlight--json">{
-	"registry": {
-		"search": [
-			"https://origami-bower-registry.ft.com",
-			"https://registry.bower.io"
-		]
-	}
-}</code>
-</pre>
+All [Origami-compliant components](/spec/v1/components) are available for installation via npm. They live in the <a href="https://registry.origami.ft.com/components">Origami Registry</a>, and are published to the <a href="https://npmjs.com/" class="o-typography-link--external">public npm registry</a>.
```
Comment (JakeChampion) [inline _tutorials/manual-build.md:97]:
```suggestion
All Origami components are available for installation via npm. They live in the <a href="https://registry.origami.ft.com/components">Origami Registry</a>, and are published to the <a href="https://npmjs.com/" class="o-typography-link--external">public npm registry</a>.
```

File: _tutorials/new-component/create-a-new-component-part-2.md [hunk 5]
```diff
@@ -162,15 +163,19 @@ The first step is to install each component we want to use via [Bower](https://b
 
 With Bower configured, we can now run:
 <pre><code class="o-syntax-highlight--bash">bower install o-colors@^5.0.0 o-spacing@^2.0.0 o-typography@^6.0.0 o-buttons@^6.0.0 --save</code></pre>
+=======
```
Comment (JakeChampion) [inline _tutorials/new-component/create-a-new-component-part-2.md:166]: A merge conflict was commited. it looks like the section below the `===` is what we want

Comment (notlee) [reply _tutorials/new-component/create-a-new-component-part-2.md:166]: Weird, I remember resolving that one -- must not have saved? good spot

File: _tutorials/new-component/create-a-new-component-part-2.md [hunk 6]
```diff
@@ -162,15 +163,19 @@ The first step is to install each component we want to use via [Bower](https://b
...
 <pre><code class="o-syntax-highlight--diff">// main.scss
 
-+@import 'o-colors/main';
-+@import 'o-spacing/main';
-+@import 'o-typography/main';
-+@import 'o-buttons/main';
++@import '@financial-times/o-colors';
++@import '@financial-times/o-spacing';
++@import '@financial-times/o-typography';
++@import '@financial-times/o-buttons';
```
Comment (JakeChampion) [inline _tutorials/new-component/create-a-new-component-part-2.md:163]: We decided to not supply a `_index.scss` in the end, we need to update this code example to use the main.scss files
```suggestion
+@import '@financial-times/o-colors/main';
+@import '@financial-times/o-spacing/main';
+@import '@financial-times/o-typography/main';
+@import '@financial-times/o-buttons/main';
```

File: _posts/2021-07-01-origami-on-npm-and-how-to-migrate.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+# Migration Guides
+
+## Using Origami via Bower
+
+All Origami components are published to the `@financial-times` namespace of NPM.
+
+1. Run the project locally in its current state, using the version of NPM the project expects right now
+1. Upgrade to the latest major versions of any Origami bower components you are using:
+    1. Confirm the project is still working correctly
+1. If the project is not using NPM version 7:
```
Comment (JakeChampion) [inline _posts/2021-07-01-origami-on-npm-and-how-to-migrate.md:47]: We found that npm 6 does work, we should update these guides to mention how to get npm 6 working if the project can't upgrade to npm 7.

This is the commit in polyfill.io that got npm 6 working correctly -- https://github.com/Financial-Times/polyfill-service/pull/2616/commits/8b9c644481c488c1a3ee5efd372e9c21408bc821

Should we wrap that script into something which projects can install and depend on rather than copy-pasting it?

Comment (JakeChampion) [reply _posts/2021-07-01-origami-on-npm-and-how-to-migrate.md:47]: Update: npm 6 could work but it is not possible to do it in an automated way. I had a chat with chee about npm 6 and we decided to not document npm 6 as we do not provide support for it

**Comment by JakeChampion** [comment]
> [ ] A blog post to describe why the spec is being deprecated, and what will happen next. Update TODO links to this.

We could write a new blog post or we could link to the [web version](https://origami.ft.com/blog/2021/06/01/newsletter/#the-origami-specification-is-no-more) of the newsletter which announced the spec deprecation. @chee, what do you think we should do for this?

**Comment by chee** [comment]
> > [ ] A blog post to describe why the spec is being deprecated, and what will happen next. Update TODO links to this.
> 
> We could write a new blog post or we could link to the [web version](https://origami.ft.com/blog/2021/06/01/newsletter/#the-origami-specification-is-no-more) of the newsletter which announced the spec deprecation. @chee, what do you think we should do for this?

that sounds good for now.

we will be writing another post at the end of Q2 with more info; we can update this section with the new link later if it seems appropriate.

File: _components/code.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+# Navigation config
+nav_display: true
+nav_label: Code
+nav_order: 12
+---
+
+# {{page.title}}
+
+The <a href="/TODO-why-we-deprecated-the-spec">Origami component specification is now deprecated</a>. How Origami components are built is under active development and may change as Origami tooling and services are updated too. This is a non-normative reference for contributing to or creating new components.
```
Comment (JakeChampion) [inline _components/code.md:14]: Need to update this link

Comment (JakeChampion) [reply _components/code.md:14]:
```suggestion
The <a href="https://origami.ft.com/blog/2021/06/01/newsletter/#the-origami-specification-is-no-more">Origami component specification is now deprecated</a>. How Origami components are built is under active development and may change as Origami tooling and services are updated too. This is a non-normative reference for contributing to or creating new components.
```

File: _manifests/origami-json.md [hunk 9]
```diff
@@ -0,0 +1,420 @@
...
+nav_label: origami.json
+nav_order: 20
+---
+
+# {{page.title}}
+
+`origami.json` is a <a href="https://www.json.org/" class="o-typography-link--external"><abbr title="JavaScript Object Notation">JSON</abbr></a> format file that is responsible for describing various aspects of an Origami project. It is under active development and subject to change, `required` properties indicate those that Origami tools or services may still rely on.
+
+<aside>
+The <a href="/TODO-why-we-deprecated-the-spec">Origami specification is now deprecated</a>. The structure of <code class="language-plaintext highlighter-rouge">origami.json</code> may change as Origami tooling and services are updated.
```
Comment (JakeChampion) [inline _manifests/origami-json.md:17]:
```suggestion
The <a href="https://origami.ft.com/blog/2021/06/01/newsletter/#the-origami-specification-is-no-more">Origami specification is now deprecated</a>. The structure of <code class="language-plaintext highlighter-rouge">origami.json</code> may change as Origami tooling and services are updated.
```

File: _tutorials/new-component/create-a-new-component-part-1.md [hunk 10]
```diff
@@ -28,20 +28,19 @@ In this tutorial we'll build an Origami component. Our example component will di
...
 
 ## The Origami Specification
 
-All Origami components **must** meet the requirements defined in the [Origami Component Specification](/spec/v1/components/). The specification covers folder structure, code standards, documentation and more. We'll refer to this document throughout this tutorial.
+The [Origami Specification](/spec/v1/) is deprecated, new components may diverge from its rules. However, some Origami tools and services continue to depend on parts of the deprecated specification so we will refer to relevant sections of the specification throughout this tutorial. Including for folder structure, code standards, documentation and more.
 
 <aside>
-If this tutorial contradicts or misses anything required in the <a href="/spec/v1/components/">Origami Component Specification</a>, the specification is correct and this tutorial is wrong. If you find a case like that please let the Origami team know so we can ensure this tutorial is always up to date.
+See our [blogpost on why the Origami specification is deprecated](TODO-why-is-spec-deprecated).
```
Comment (JakeChampion) [inline _tutorials/new-component/create-a-new-component-part-1.md:43]:
```suggestion
See our [blogpost on why the Origami specification is deprecated](https://origami.ft.com/blog/2021/06/01/newsletter/#the-origami-specification-is-no-more).
```

**Review by JakeChampion — APPROVED** [review]

File: _tutorials/manual-build.md [hunk 11]
```diff
@@ -257,7 +238,7 @@ To choose a brand other than the default "master" brand, set the `$o-brand` <abb
 
 To see this in action we can set our brand to "internal":
 <pre><code class="o-syntax-highlight--scss">$o-brand: "internal"; // Set brand before anything else.
-@import 'o-colors/main';
+@import '@financial-times/o-colors';
```
Comment (notlee) [inline _tutorials/manual-build.md:241]:
```suggestion
@import '@financial-times/o-colors/main';
```

**Comment by notlee** [comment]
I can't review this as I opened the PR, but I approve
